addons/mail/static/src/web/chatter.js

```diff
@@ -196,6 +202,14 @@ export class Chatter extends Component {
             },
             () => [this.state.thread?.isLoadingAttachments]
         );
+        useEffect(
+            () => {
+                if (this.state.reloadParentView) {
+                    this.reloadParentView();
+                }
+            },
+            () => [this.state.reloadParentView]
+        );
```

Comment:
It doesn't look necessary to make a `state.reloadParentView` and have this `useEffect`.
You could use `this.reloadParentView()` like before, but make this function available in `useChildSubEnv`. In documents override, it just has to `this.env.reloadParentView()`

Reply:
This setup made sense since we wished to use flag, but if we omit that, then we can also use `this.props.reloadParentView()` in our patch, that would remove the need for adding it in env unless we wish to remove that prop as well

Reply:
Either I miss something or using `this.state.reloadParentView` is just a complicated way to just `this.reloadParentView()`.

Using a flag in `state`, so that it triggers a `useEffect` that actually `reloadParentView()`... This looks complicated for the sake of being complicated.

The only part that may require something like this is if you have 2 code triggering the flag and you rely on `useEffect` to result in a single `reloadParentView()` rather than 2... Is it what you try to solve? If not, then I see no point to make the code more complicated than necessary

Reply:
Alright, I understand the reason behind removing the flag.
I've pushed the changes as suggested. 
Now, since we added the method in `env`, should we remove the method from `props` and adapt accordingly? or keep them both for older patches to work properly?

Reply:
We could, but it's best to do that in `master`. This is a fix in 16.3 so we should be cautious not to break community code like OCA modules, which could rely on `this.props.reloadParentView`.

Reply:
Alright in that case, I have implemented it and have updated both branches accordingly